app/src/main/java/org/ergoplatform/android/wallet/SaveWalletFragmentDialog.kt

```diff
@@ -108,6 +108,7 @@ class SaveWalletFragmentDialog : FullScreenFragmentDialog(), PasswordDialogCallb
     }
 
     private fun saveToDb(encType: Int, secretStorage: ByteArray) {
+        // TODO avoid using mnemonic here, store and use publicAddress directly
```

Comment:
Is this added by intention?

Asking because what is described is exactly what happens here: the public address is stored to the db for future use (unencrypted), the encrypted mnemonic is saved to db.

It is possible to use the formerly calculated address from `onCreate` as well, however, this won't be more secure. The mnemonic is stored in memory in args (unencrypted) anyway as long as it is encrypted and saved to db and the dialog closes.

Reply:
Yes, I wrote it to pay attention. It is not a show stopper.
A better approach is to use the SecretString class from Appkit for mnemonic and erase it once it is no longer needed, to limit its lifetime.

Reply:
SecretString can't be used as an argument for the dialog because the arguments are handled by Android. It needs to do so because under certain circumstances (low memory or some configuration changes like changing the system language in the settings) the system will safe the arguemnts, destroy the current dialog instance and recreate it with the formerly used arguments. This does not work with own classes (or more precisely, own classes will be serialized and thus the secret string is not secret). Ways to work around this are possible, but have other caveats on their own which I think are more problematic than having the mnemonic as a String here (using a static won't get reset on dialog destruction and might be accessible by other processes reusing the JVM instance, using a ViewModel might lose the saved mnemonic under some circumstances which is more secure but very unreliable for the user). In short, I am aware that we are not perfect here but as good as we can get. I will extend the comment so others can see why it is done this way and maybe someone comes up with a better approach in the future.